src/pages/index.js

```diff
@@ -66,57 +49,155 @@ const cardGallerySelector = ".gallery__grid";
 //access the card template
 const cardSelector = "#card-template";
 
-//function to create card
 let place = "append";
+const cardHolder = [];
+//function to create card
 const createCard = (item) => {
-  const card = new Card(item, cardSelector, handleImageClick);
+  const card = new Card({
+    data: item,
+    cardSelector: cardSelector,
+    handleImageClick: handleImageClick,
+    handleTrashClick: () => {
+      const cardId = card.getId();
+      confirmModal.open(card, cardId);
+    },
+    toggleLike: () => {
+      const cardId = card.getId();
+      const isLiked = card.getLikeStatus();
+      if (isLiked) {
+        newApi.addLike({ cardId: cardId, isLiked: isLiked }).then((res) => {
+          console.log(`liking: ${cardId}`);
+          console.log(res);
+        });
+      } else {
+        newApi.removeLike({ cardId: cardId, isLiked: isLiked }).then((res) => {
+          console.log(`disliking: ${cardId}`);
+          console.log(res);
+        });
+      }
+    },
+  });
   const cardElement = card.getView();
-
   section.addItem(cardElement, place);
+
+  return { card, cardId: card.getId() };
 };
 const section = new Section(
-  { items: initialCards, renderer: createCard },
+  { items: cardHolder, renderer: createCard },
   cardGallerySelector
 );
-section.renderItems();
+
+const handleDeleteConfirm = (currentCard, cardId) => {
+  const isNotSubmitted = !confirmModal.getIsSubmitted();
+  console.log(`not submitted: ${isNotSubmitted}`);
+  if (!confirmModal.getIsSubmitted()) {
+    console.log("no submission");
+  } else {
+    newApi.deleteCard({ cardId: cardId }).then((res) => {
+      console.log(res, cardId);
+    });
+    //remove card from page
+    currentCard.removeCard();
```

Comment:
Needs correction: you should update the ui only after the successful api response in `then` block